cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PullCommand.java

```diff
@@ -206,6 +211,25 @@ public void execute() {
         }
     }
 
+    private void pullDependencyPackages(String orgName, String packageName, String version) {
+        Path ballerinaUserHomeDirPath = ProjectUtils.createAndGetHomeReposPath();
+        Path remoteRepositoryDirPath = ballerinaUserHomeDirPath.resolve(ProjectConstants.REPOSITORIES_DIR)
+                .resolve(ProjectConstants.CENTRAL_REPOSITORY_CACHE_NAME);
+        Path packagePathInBalaCache = remoteRepositoryDirPath.resolve(ProjectConstants.BALA_DIR_NAME)
+                .resolve(orgName).resolve(packageName);
+        Path balaPath = packagePathInBalaCache.resolve(version);
+        String ballerinaShortVersion = RepoUtils.getBallerinaShortVersion();
+        Path cacheDir = remoteRepositoryDirPath.resolve(ProjectConstants.CACHES_DIR_NAME + "-" + ballerinaShortVersion);
+
+        ProjectEnvironmentBuilder defaultBuilder = ProjectEnvironmentBuilder.getDefaultBuilder();
+        defaultBuilder.addCompilationCacheFactory(new FileSystemCache.FileSystemCacheFactory(cacheDir));
+        BalaProject balaProject = BalaProject.loadProject(defaultBuilder, balaPath);
+
+        // getResolution pulls all dependencies of the pulled package
+        PackageCompilation packageCompilation = balaProject.currentPackage().getCompilation();
+        packageCompilation.getResolution();
```

Comment:
Calling the compilation will internally call the resolution as well. So we can remove this line. Additionally, we should check compilation diagnostics and if there exists any, print here and exit. Else, go ahead and generate caches using `JBallerinaBackend`